tsconfig.json

```diff
@@ -2,8 +2,8 @@
 	"compileOnSave": true,
 	"compilerOptions":
 	{
-		"lib": [ "es2019" ],
-		"target": "es2020",
+		"lib": [ "es2017" ],
+		"target": "es2017",
```

Comment:
I still don't know why this is needed. In your previous PR you changed this due to the usage of `array.flatten()` (not available in Node 10 AFAIR). There is no `array.flatten()` here. Why is this change required? Obviously CI is working and it tests Node 10, 12 and 14:

- https://travis-ci.com/github/versatica/mediasoup
- https://travis-ci.com/github/versatica/mediasoup/jobs/347408876#L1423

Reply:
It doesn't matter whether you are using something that is forbidden or not now, the issue is that you can introduce breaking change at any point and compiler will not stop you (yes, Rust had influence on me in this direction).

You can put `??` operator in the code now and only Node.js 14 will be able to handle it, while compiler will tell nothing. So target should represent actual target you expect code to run on. Same about libraries.

Reply:
I see, the problem is the `target` and not `lib`. And since we are supposed to support Node 10 we must produce code valid for Node 10, and that means  `target: "es2017"`, right? However:

* Why not `target: "es2018"`? It seems to be fully supported by Node 10:
  - https://stackoverflow.com/questions/51716406/typescript-tsconfig-settings-for-node-js-10

* Why must be `lib` also `es2017` or `es2018`? Why not `es2019` or `es2020` so I can use `??` eventually?

Reply:
I don't mind about `es2018` target, but didn't check if everything is supported in TS. If it is feel free to edit it.
`lib` is not just syntax, but also things like JS standard library. If you specify `es2019` things like that `flatten()`/`flatMap()` will be left as is and will not be polyfilled for you. So it will break in Node.js 10.

Reply:
`??` was not polyfilled either if I remember correctly.

Reply:
Tested right now with:

```
"lib": [ "es2019" ],
"target": "es2018",
```

TypeScript code:

```ts
let foo: string | undefined;

console.warn('foo.length:', foo?.length ?? 'UNSET');
```

Compiled code:

```js
let foo;
console.warn('foo.length:', (_a = foo === null || foo === void 0 ? void 0 : foo.length) !== null && _a !== void 0 ? _a : 'UNSET');
```

Reply:
Yet, it will not work with JS API, so... https://travis-ci.com/github/nazar-pc/mediasoup/jobs/345707263

Reply:
That's unrelated IMHO. There you are using `array.flatten()` which does not exist in Node 10, that's the problem. TypeScript does not even try to polyfill it.

Reply:
> TypeScript does not even try to polyfill it.

Right, but with `"lib": [ "es2017" ],` it wouldn't even compile, isn't than nice to have?

Reply:
But we are not using `array.flatten()` in the code :)

Reply:
That is not the point. The point is that compiler proactively prevents you to even putting things that don't work into the repository.

Reply:
So do you mean that if we use `lib: [ "es2018" ]` instead of `es2020`, then the compiler will complain if I try to use `array.flatten()`?

Reply:
Exactly

Reply:
Good reason. Can we then use `es2018` for lib and target? As I pointed out before, it's supposed to be perfectly supported by Node 10 (no need to use es2017 anywhere).

Reply:
I think so

Reply:
Cool. Can you update your PR with just that, please?

Reply:
Done